tests/jam_test.cpp

```
@@ -0,0 +1,169 @@
+  // A sender whose hat is a different pattern (two steps, a split), at 120 bpm in A minor.
+  const engine::State sender = engine::decode(kSenderLoop, app::kit()).state;
+  uint8_t msg[io::kMessageCapacity];
+  const int n = io::format_track(sender, app::kit(), engine::index_of(Pad::hat), msg);
+  hal_fake::push_midi(msg, n);
+  w.run_for(kSecond / 10);
+
+  // The hat's pattern became the sender's; the receiver's tempo, key and the hat's own
+  // level stayed put (patterns travel, knobs and globals do not).
+  const engine::Track after = engine::track_of(w.state(0), Pad::hat);
+  CHECK(after.step_count == engine::track_of(sender, Pad::hat).step_count);
+  CHECK(after.step_count != before.step_count);
```

Cppcheck (2.21.0)

[style] The function 'skip_timer_until' is never used.

(unusedFunction)